WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift

```diff
@@ -1016,46 +1012,47 @@ import Combine
             return
         }
 
-        syncContext.perform { [weak self] in
-            guard let topicInContext = (try? self?.syncContext.existingObject(with: topic.objectID)) as? ReaderAbstractTopic else {
-                DDLogError("Error: Could not retrieve an existing topic via its objectID")
-                return
-            }
-
-            let objectID = topicInContext.objectID
+        let objectID = topic.objectID
 
-            let successBlock = { [weak self] (count: Int, hasMore: Bool) in
-                DispatchQueue.main.async {
-                    if let strongSelf = self {
-                        if strongSelf.recentlyBlockedSitePostObjectIDs.count > 0 {
-                            strongSelf.recentlyBlockedSitePostObjectIDs.removeAllObjects()
-                            strongSelf.updateAndPerformFetchRequest()
-                        }
-                        strongSelf.updateLastSyncedForTopic(objectID)
+        let successBlock = { [weak self] (count: Int, hasMore: Bool) in
+            DispatchQueue.main.async {
+                if let strongSelf = self {
+                    if strongSelf.recentlyBlockedSitePostObjectIDs.count > 0 {
+                        strongSelf.recentlyBlockedSitePostObjectIDs.removeAllObjects()
+                        strongSelf.updateAndPerformFetchRequest()
                     }
-                    success?(hasMore)
+                    strongSelf.updateLastSyncedForTopic(objectID)
                 }
+                success?(hasMore)
             }
+        }
 
-            let failureBlock = { (error: Error?) in
-                DispatchQueue.main.async {
-                    if let error = error {
-                        failure?(error as NSError)
-                    }
+        let failureBlock = { (error: Error?) in
```

Comment:
This `Optional<Error>` is surprising. However, it's definitely out of scope for this PR to tackle it.

I had a quick look down the call chain, and I couldn't find an instance where the error received upstream was nil. It might be a simple matter of changing the type and let the compiler tell us what to do next.

Reply:
Never mind that... I kept looking and realized there's Objective-C code involved so addressing this might involve more steps than I thought, including adding nullability annotations and understanding whether the Objective-C code can actually run into a case where the error object is `nil`.